### DISCUSSION TITLE
csv file and gpx file in python

### DISCUSSION THREAD

**Discussion by OutTF** [duckdb/duckdb#6750] [Q&A]
Hello, I am trying to read my csv file and write it into gpx file.

I could read it but when I try to write it did only write the last line.

I would like to write each line one by one.

This is my program :

import csv
from csv import DictReader

with open('info.csv', 'r', newline='') as read_obj:
    csv_dict_reader = DictReader(read_obj)
    
    #for row in csv_dict_reader:
        #print(row['name'], row['latitude'], row['longitude'])

    x = int(input("How many lines: "))
    
    while x > 0 :
    
        filepath = ""
        filename = "data.gpx"
        for row in csv_dict_reader:
            I1 = "<wpt lat=" '"' + row['latitude'] + '"' " lon=" '"' + row['longitude']+ '"' ">"
            I2 = "  <name><![CDATA[" + row['name'] + "]]></name>"
            I3 = "</wpt>"
            
        if x > 0 :
            try:
                d = open(filepath+filename,"a")
                print("File", filename, "is open")

                try:
                    d.write("\n" + I1+ "\n" + I2 + "\n" + I3+ "\n")
                    print("Data written to file")

                except:
                    print ("Write error")

                try:
                    d.close()
                    print("File", filename, "is closed")    

                except:
                    print("An error occurred while closing the file")

            except:
                print("Error: file", filename, "could not be opened")

            x = x - 1

read_obj.close()

 I am new at programming.

**Comment by Mytherin** [comment]
I would try asking your question in the [Python StackOverflow](https://stackoverflow.com/questions/tagged/python) - this question is unrelated to DuckDB.
